applications/GeoMechanicsApplication/tests/cpp_tests/test_time_loop_executor.cpp

```diff
@@ -0,0 +1,359 @@
+
+    KRATOS_EXPECT_DOUBLE_EQ(std::accumulate(increments.begin(), increments.end(), 0.), step_states.back().time);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GetThreeCyclesWhenThreeCyclesAreNeeded, KratosGeoMechanicsFastSuite)
+{
+    TimeLoopExecutor executor;
+    const auto wanted_num_of_cycles_per_step = 3;
+    executor.SetTimeIncrementor(std::make_unique<FixedCyclesTimeIncrementor>(wanted_num_of_cycles_per_step));
+    executor.SetSolverStrategyTimeIncrementor(std::make_unique<DummySolverStrategy>(TimeStepEndState::ConvergenceState::converged));
+    const auto step_states = executor.Run(TimeStepEndState{});
+
+    // The test time incrementor assumes fixed time increments of 0.5 and an end time of 1.0
+    KRATOS_EXPECT_EQ(2, step_states.size());
+    KRATOS_EXPECT_DOUBLE_EQ(0.5, step_states[0].time);
+    KRATOS_EXPECT_EQ(wanted_num_of_cycles_per_step, step_states[0].num_of_cycles);
+    KRATOS_EXPECT_DOUBLE_EQ(1.0, step_states[1].time);
+    KRATOS_EXPECT_EQ(wanted_num_of_cycles_per_step, step_states[1].num_of_cycles);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GetFiveCyclesWhenFiveCyclesAreNeeded, KratosGeoMechanicsFastSuite)
+{
+    TimeLoopExecutor executor;
+    const auto wanted_num_of_cycles_per_step = 5;
+    executor.SetTimeIncrementor(std::make_unique<FixedCyclesTimeIncrementor>(wanted_num_of_cycles_per_step));
+    executor.SetSolverStrategyTimeIncrementor(std::make_unique<DummySolverStrategy>(TimeStepEndState::ConvergenceState::converged));
+    const auto step_states = executor.Run(TimeStepEndState{});
+
+    // The test time incrementor assumes fixed time increments of 0.5 and an end time of 1.0
+    KRATOS_EXPECT_EQ(2, step_states.size());
+    KRATOS_EXPECT_DOUBLE_EQ(0.5, step_states[0].time);
+    KRATOS_EXPECT_EQ(wanted_num_of_cycles_per_step, step_states[0].num_of_cycles);
+    KRATOS_EXPECT_DOUBLE_EQ(1.0, step_states[1].time);
+    KRATOS_EXPECT_EQ(wanted_num_of_cycles_per_step, step_states[1].num_of_cycles);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(ExpectOneCyclePerStepWhenUsingAPrescribedTimeIncrementor, KratosGeoMechanicsFastSuite)
+{
+    TimeLoopExecutor executor;
+    const auto increments = std::vector<double>{0.1, 0.2, 0.3, 0.4};
+    executor.SetTimeIncrementor(std::make_unique<PrescribedTimeIncrementor>(increments));
+    executor.SetSolverStrategyTimeIncrementor(std::make_unique<DummySolverStrategy>(TimeStepEndState::ConvergenceState::converged));
+    const auto step_states = executor.Run(MakeConvergedStepState());
+
+    KRATOS_EXPECT_EQ(increments.size(), step_states.size());
+    KRATOS_EXPECT_TRUE(std::all_of(step_states.begin(), step_states.end(),
+                       [](const auto& state) {return state.num_of_cycles == 1;}))
+    auto time = 0.0;
+    for (auto index = std::size_t{0}; index < increments.size(); ++index) {
+        time += increments[index];
+        KRATOS_EXPECT_DOUBLE_EQ(time, step_states[index].time);
+    }
+}
+
+KRATOS_TEST_CASE_IN_SUITE(ExpectEndTimeToBeSetAfterRunningAStep, KratosGeoMechanicsFastSuite)
```

Comment:
I think it would be clearer to rename ExpectEndTimeToBeSetAfterRunningAStep to ExpectEndTimeToBeSetAfterRunningATimeLoop

Reply:
Test name adapted.